xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrMediaTypeFacet.xml

```diff
@@ -151,15 +151,15 @@
               $facetRequestValuesBackup.size())))
             #set ($discard = $facetRequestValues.removeAll($category.mediaTypes))
           #end
-          &lt;div class="mediaTypeCategory"&gt;#displaySearchFacetValue($category {} 'displaySearchFacetValue_mediaType')&lt;/div&gt;
+          &lt;div id="mediatype-$stringtool.substringBefore($category.name, '\')" class="mediaTypeCategory"&gt;#displaySearchFacetValue($category {} 'displaySearchFacetValue_mediaType' true)&lt;/div&gt;
```

Comment:
Shouldn't this be a `/`? Or is `$category.name` not something like `image/png`?

Reply:
Examples of the value in `$category.name` :
> (text\/* application\/xml* application\/javascript* application\/ecmascript* application\/json* application\/x\-sh* *\+xml)

>  (image\/*)

Here, taking the first substring is enough to get an unique ID, and AFAIK, we avoid any space (that would break the id parameter). It was not very clean (giving IDs like `mediatype-(image` ), so I cleaned it up in https://github.com/xwiki/xwiki-platform/pull/2311/commits/adabed9cbeb5208c5280e889848924b4f41700d2 :+1: